### ISSUE TITLE
[java] UnnecessaryFullyQualifiedName: false positive for nested inner class defining method with same name as static import

### ISSUE THREAD

**Issue by cobratbq** [pmd/pmd#1316]
**Affects PMD Version:** PMD 6.5.0 running using maven plug-in 3.10.0 (also present in version 6.6.0).

**Rule:** UnnecessaryFullyQualifiedName (category/java/codestyle.xml/UnnecessaryFullyQualifiedName)

**Description:**
The rule correctly identifies that a static import is defined for `sessionStatusChanged`. The implementation does not identify that the method the find is used in is defined as `sessionStatusChanged` as well. Therefore, calling the static import name would result in a recursive method call instead of a reference to the static import. It is necessary to refer to the method call by class name.

**Code Sample demonstrating the issue:**

```
private final OtrEngineListener slaveSessionsListener = new OtrEngineListener() {

    // FIXME temporarily suppress warning due to false-positive in use of existing static import.
    @SuppressWarnings("PMD.UnnecessaryFullyQualifiedName")
    @Override
    public void sessionStatusChanged(@Nonnull final SessionID sessionID, @Nonnull final InstanceTag receiver) {
        OtrEngineListenerUtil.sessionStatusChanged(duplicate(listeners), sessionID, receiver);
    }

    // ...
};
```

**Running PMD through:** Maven

**Comment by jsotuyod** [comment]
This issue is no longer reproducible under PMD 7.0.0
